pkg/cmd/infra/router/router.go

```diff
@@ -26,8 +27,24 @@ This command launches a router connected to your OpenShift master. The router li
 created by users and keeps a local router configuration up to date with those changes.`
 )
 
+// routerConfig represents a container for all other router config types.
+// Any global config belongs on this level.
+type routerConfig struct {
+	//Config is global flags that are passed to all client commands that need to connect to master
+	Config *clientcmd.Config
+
+	//Type is the type of plugin being created
+	Type string
+
+	//TemplateRouterConfig holds config items for template router plugins
+	TemplateRouterConfig *templateRouterConfig
```

Comment:
not sure the purposes of making these a pointer if they will always be initialized in `newEmptyConfig`.  Incidentally, if they aren't pointers we don't need `newEmptyConfig`.  I don't have a preference, just curious about the decision